opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -0,0 +1,299 @@
+   * @return A list of format names and user defined formats.
+   */
+  private List<String> getFormatList() {
+    String format = strip8Prefix(formatString);
+    List<String> patterns = splitCombinedPatterns(format);
+    return patterns;
+  }
+
+
+  /**
+   * Retrieves a list of named formatters defined by OpenSearch.
+   * @return a list of DateFormatters that can be used to parse a Date/Time/Timestamp.
+   */
+  public List<DateFormatter> getAllNamedFormatters() {
+    return getFormatList().stream()
+        .filter(formatString -> FormatNames.forName(formatString) != null)
+        .map(DateFormatter::forPattern).collect(Collectors.toList());
+  }
+
+  /**
+   * Retrieves a list of custom formatters defined by the user.
+   * @return a list of DateFormatters that can be used to parse a Date/Time/Timestamp.
+   */
+  public List<DateFormatter> getAllCustomFormatters() {
+    return getFormatList().stream()
+        .filter(formatString -> FormatNames.forName(formatString) == null)
+        .map(DateFormatter::forPattern).collect(Collectors.toList());
+  }
+
+  /**
+   * Retrieves a list of named formatters that format for dates.
+   *
+   * @return a list of DateFormatters that can be used to parse a Date.
+   */
+  public List<DateFormatter> getDateNamedFormatters() {
+    return getFormatList().stream()
+        .filter(formatString -> {
+          FormatNames namedFormat = FormatNames.forName(formatString);
+          return SUPPORTED_NAMED_DATE_FORMATS.contains(namedFormat);
+        })
+        .map(DateFormatter::forPattern).collect(Collectors.toList());
+  }
+
+  /**
+   * Retrieves a list of named formatters that format for Times.
+   *
+   * @return a list of DateFormatters that can be used to parse a Time.
+   */
+  public List<DateFormatter> getTimeNamedFormatters() {
+    return getFormatList().stream()
+        .filter(formatString -> {
+          FormatNames namedFormat = FormatNames.forName(formatString);
+          return SUPPORTED_NAMED_TIME_FORMATS.contains(namedFormat);
+        })
+        .map(DateFormatter::forPattern).collect(Collectors.toList());
+  }
+
+  private ExprCoreType getExprTypeFromFormatString(String formatString) {
+    if (formatString.isEmpty()) {
+      // TODO: check the default formatter - and set it here instead of assuming that the default
+      // is always a timestamp
+      return ExprCoreType.TIMESTAMP;
+    }
+
+    List<DateFormatter> namedFormatters = getAllNamedFormatters();
+
+    if (namedFormatters.isEmpty()) {
+      return ExprCoreType.TIMESTAMP;
+    }
+
+    if (!getAllCustomFormatters().isEmpty()) {
+      // TODO: support custom format in <issue#>
+      return ExprCoreType.TIMESTAMP;
+    }
+
+    // if there is nothing in the dateformatter that accepts a year/month/day, then
+    // we can assume the type is strictly a Time object
+    if (namedFormatters.size() == getTimeNamedFormatters().size()) {
```

Comment:
Consider having a more reliable check: amounts of named time and date formatters could be same. Or leave a comment for future you.

Reply:
Will be included in a follow up PR including user defined formats